<a href="https://colab.research.google.com/github/decastrosantos/BankValley_Proyecto/blob/main/BankValley_Proyecto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import random

from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier, export_text
from sklearn.metrics import classification_report

# Cargar el dataset > Dataset ficticio simulado para scoring hipotecario
data = pd.read_csv("bankvalley_internal_clients.csv")
data.head()

# Variables independientes y dependiente
X = data[[
    "age", "monthly_income", "is_paying_rent", "avg_rent_payment",
    "web_visits_hipoteca", "used_simulator", "products_owned", "tenure_years"
]]
y = data["score_target"]

# División train/test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Modelo de árbol de decisión
clf = DecisionTreeClassifier(max_depth=4, random_state=42)
clf.fit(X_train, y_train)

# Predicciones
y_pred = clf.predict(X_test)

# Reporte de métricas
report = classification_report(y_test, y_pred, output_dict=True)

# Reglas del árbol para interpretación
tree_rules = export_text(clf, feature_names=list(X.columns))




✅ Modelo de Scoring entrenado con Árbol de Decisión

🧠 Reglas aprendidas por el modelo

1. Si visitas sección de hipotecas ≤ 1 → No interesado (Clase 0)
2. Si visitas > 1:
   a. ¿No paga alquiler? → No interesado
   b. ¿Paga alquiler?
      i. ¿Ingreso ≤ €2.034? → No interesado
      ii. ¿Ingreso > €2.034? → Interesado (Clase 1)
      

Esto significa que los factores más relevantes para identificar un potencial cliente de hipoteca son:

🧭 Comportamiento digital (visitas a hipotecas)

🏠 Si paga alquiler

💸 Ingresos mensuales      

In [ ]:
tree_rules, report

('|--- web_visits_hipoteca <= 1.50\n|   |--- class: 0\n|--- web_visits_hipoteca >  1.50\n|   |--- is_paying_rent <= 0.50\n|   |   |--- class: 0\n|   |--- is_paying_rent >  0.50\n|   |   |--- monthly_income <= 2034.00\n|   |   |   |--- class: 0\n|   |   |--- monthly_income >  2034.00\n|   |   |   |--- class: 1\n',
 {'0': {'precision': 0.9852941176470589,
   'recall': 1.0,
   'f1-score': 0.9925925925925926,
   'support': 67.0},
  '1': {'precision': 1.0,
   'recall': 0.9565217391304348,
   'f1-score': 0.9777777777777777,
   'support': 23.0},
  'accuracy': 0.9888888888888889,
  'macro avg': {'precision': 0.9926470588235294,
   'recall': 0.9782608695652174,
   'f1-score': 0.9851851851851852,
   'support': 90.0},
  'weighted avg': {'precision': 0.9890522875816994,
   'recall': 0.9888888888888889,
   'f1-score': 0.9888065843621399,
   'support': 90.0}})


📊 Métricas del modelo (con datos de test)

Métrica | Valor

Accuracy | 98.9%

F1 Score (clientes interesados) | 97.8%

Precision (interesados) | 100%

Recall (interesados) | 95.7%


